graylog2-server/src/main/java/org/graylog/plugins/views/search/rest/scriptingapi/mapping/AggregationTabularResponseCreator.java

```diff
@@ -96,6 +114,30 @@ private static List<List<Object>> getDatarows(final AggregationRequestSpec searc
                 .collect(Collectors.toList());
     }
 
+    private Stream<Object> decorateGroupings(ImmutableList<String> keys, AggregationRequestSpec searchRequestSpec, Set<FieldDecorator> decorators, SearchUser searchUser) {
+        List<Object> result = new ArrayList<>();
+        for(int i = 0; i < keys.size(); i++) {
+            final String value = keys.get(i);
+            final RequestedField field = searchRequestSpec.groupings().get(i).requestedField();
+            final Object decorated = decorate(decorators, field, value, searchUser);
+            result.add(decorated);
+        }
+        return result.stream();
+    }
+
+    private Object decorate(Set<FieldDecorator> decorators, RequestedField field, Object val, SearchUser searchUser) {
+        final Optional<Object> decorated = decorators.stream()
+                .filter(d -> d.accept(field))
+                .findFirst()
+                .map(d -> d.decorate(field, val, searchUser));
+
+        if (decorated.isEmpty() && field.hasDecorator()) {
+            throw new IllegalArgumentException(StringUtils.f("Unsupported decorator '%s' on field '%s'", field.decorator(), field.name()));
```

Comment:
Do the API user need to know/be aware of the concept of decorator at all?
If they use "stream.abracadabra" in the API, they should rather receive more general error message, like "cannot provide 'abracadabra' property for stream", it would be nice if it could be followed by "valid properties : id, name, title, whatever", automatically taken from the list of decorators.

Reply:
I think providing the list of the properties would destroy the simplicity of the accept method and would require way more complicated logic. But the phrasing should be changed, will do that. Thanks1